File: api/internal/model/device.go
package model

import "time"

// SrvlogDeviceSummary holds aggregated information for a single device (hostname).
type SrvlogDeviceSummary struct {
	Hostname          string          `json:"hostname"`
	LastSeenAt        *time.Time      `json:"last_seen_at"`
	TotalCount        int64           `json:"total_count"`
	CriticalCount     int64           `json:"critical_count"`
	SeverityBreakdown []SeverityCount `json:"severity_breakdown"`
	TopMessages       []TopMessage    `json:"top_messages"`
	CriticalLogs      []SrvlogEvent   `json:"critical_logs"`
}

// TopMessage holds a normalized message pattern with its count and a sample.
// Severity is an int (RFC 5424 numeric level 0-7) because srvlog events use
// integer severity codes. Compare with AppLogTopMessage.Level which is a
// string ("DEBUG","INFO","WARN","ERROR","FATAL") because applog events use
// freeform text levels. This asymmetry reflects the different source protocols.
type TopMessage struct {
	Pattern       string    `json:"pattern"`
	Sample        string    `json:"sample"`
	Count         int64     `json:"count"`
	LatestID      int64     `json:"latest_id"`
	LatestAt      time.Time `json:"latest_at"`
	Severity      int       `json:"severity"`
	SeverityLabel string    `json:"severity_label"`
}

File: frontend/src/types/device.ts
import type { LevelCount, SeverityCount } from '@/types/stats'
import type { SrvlogEvent } from '@/types/srvlog'
import type { AppLogEvent } from '@/types/applog'

export interface TopMessage {
  pattern: string
  sample: string
  count: number
  latest_id: number
  latest_at: string
  severity: number
  severity_label: string
}

export interface ActivityBucket {
  time: string
  count: number
}

export interface DeviceSummary {
  hostname: string
  fromhost_ip: string
  last_seen_at: string | null
  total_count: number
  critical_count: number
  severity_breakdown: SeverityCount[]
  top_messages: TopMessage[]
  critical_logs: SrvlogEvent[]
  activity: ActivityBucket[]
}

export interface DeviceSummaryResponse {
  data: DeviceSummary
}

export interface AppLogTopMessage {
  pattern: string
  sample: string
  count: number
  latest_id: number
  latest_at: string
  level: string
}

export interface AppLogDeviceSummary {
  host: string
  last_seen_at: string | null
  total_count: number
  error_count: number
  level_breakdown: LevelCount[]
  top_messages: AppLogTopMessage[]
  error_logs: AppLogEvent[]
  activity: ActivityBucket[]
}

export interface AppLogDeviceSummaryResponse {
  data: AppLogDeviceSummary
}
